core/controllers/admin.py

```diff
@@ -656,6 +658,453 @@ def _generate_dummy_explorations(
         else:
             raise Exception('Cannot generate dummy explorations in production.')
 
+    def _generate_sample_opportunities(
+            self, num_sample_ops,
+            num_sample_interactions_per_opportunity,
+            should_submit_suggestions):
+        """Generates and publishes the given number of sample opportunities.
+
+        Args:
+            num_sample_ops: int. Number of sample opportunities to
+                generate.
+            num_sample_interactions_per_opportunity: int. Number of sample
+                interactions to generate per opportunity.
+            should_submit_suggestions: boolean.
+                Track whether suggestions should be submitted.
+
+        Raises:
+            Exception. Environment is not DEVMODE.
+        """
+        if constants.DEV_MODE:
+            if feconf.ROLE_ID_CURRICULUM_ADMIN not in self.user.roles:
+                raise Exception(
+                    'User does not have enough rights to generate data.')
+            if story_services.does_story_exist_with_url_fragment(
+                'sample-story-title'):
+                self._generate_sample_opportunities_with_existing_topic(
+                    num_sample_ops,
+                    num_sample_interactions_per_opportunity,
+                    should_submit_suggestions,
+                    'sample-story-title')
+            else:
+                self._generate_sample_opportunities_without_existing_topic(
+                    num_sample_ops,
+                    num_sample_interactions_per_opportunity,
+                    should_submit_suggestions)
+
+    def _generate_sample_opportunities_without_existing_topic(
```

Comment:
Generally order functions in the order that they're used, so this should be below `_generate_sample_opportunities_with_existing_topic`.